raft-tech/TANF-app
# Issue 1053: Update django-admin-508 for a11y fixes

### Summary of Changes

This PR updates `django-adming-508` so as to fix several outstanding a11y issues in Django Admin.

List of fixes: <https://github.com/raft-tech/django-admin-508/blob/main/CHANGELOG.md#003-2021-07-21>

_Addresses issue_ #1053

**Completed Tasks:**
- [x] Evaluate a11y fixes coming from fork 
- - [x] Add outstanding contrast fixes if needed (See Task 4 on hack.md)
- [x] Add dynamic labels to checkbox elements (See Task 1 on hack.md)
- [x] Ensure errors and other state changes are read by screenreaders (See Task 3 on hack.md)
- [x] Make heading styles and associated markup consistent across Django Admin (See Task 6 on hack.md)
- [x] Retain the tooltips, implement accessible tooltip ARIA markup (See Task 2 on hack.md)

**Supporting Documentation:**
[https://hackmd.io/@mreiter/ryuHXhbuu](https://hackmd.io/@mreiter/ryuHXhbuu
) Contains full task list for a11y fixes in support of #892 

### How to Test
_List the steps to test the PR_

1. Log into Django Admin
2. Confirm the fixes noted in the tasks above

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

	- [x] Django admin is fully visually accessible (e.g. color, contrast) to at least a WCAG 2.0 AA standard.
	- [x] Django admin is more accessible to screen reader only users (markup conventions, all alerts / state / value updates are read) 

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

Does not apply; this PR introduces only a dependency update, and no changes to backend or frontend application code.

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

Does not apply, no code changes included.

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

Does not apply, existing dependency is being updated to a newer version.

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

	- No issues found
